forge-ai/src/main/java/forge/ai/ComputerUtilCombat.java

```diff
@@ -2046,8 +2046,15 @@ public static Map<Card, Integer> distributeAIDamage(final Player self, final Car
             // TODO sort remaining tramplers for DamageDone triggers
         }
 
-        if (block.size() == 1) {
-            final Card blocker = block.getFirst();
+        // Order the combatants in preferred order in case legacy ordering is disabled
+        final boolean legacyOrderCombatants = self.getGame().getRules().hasOrderCombatants();
+        final CardCollection orderedBlockers = new CardCollection(block);
+        if (!legacyOrderCombatants) {
+            ComputerUtilCard.sortByEvaluateCreature(orderedBlockers); // assume sorted in case the legacy option is enabled
```

Comment:
reuse `AiBlockController.orderBlockers`

Reply:
Done :+1:

Reply:
wrong implementation:
that function isn't void for a reason

Reply:
Ah, lol, I was wondering why it actually worked the way it does, looks like it modifies orderedBlockers as a side effect due to the evaluate call, but then does more stuff and returns the final version as a result. Modified :+1: